app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageAiManager.kt

```diff
@@ -0,0 +1,246 @@
+ * Copyright (C) 2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.newMessage
+
+import android.animation.FloatEvaluator
+import android.animation.ValueAnimator
+import android.app.Activity
+import android.content.Context
+import android.transition.Slide
+import android.transition.TransitionManager
+import android.view.ViewGroup.FOCUS_BEFORE_DESCENDANTS
+import android.view.ViewGroup.FOCUS_BLOCK_DESCENDANTS
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import com.infomaniak.lib.core.utils.hideKeyboard
+import com.infomaniak.lib.core.utils.safeNavigate
+import com.infomaniak.mail.MatomoMail.trackAiWriterEvent
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.LocalSettings
+import com.infomaniak.mail.data.models.FeatureFlag
+import com.infomaniak.mail.data.models.ai.AiPromptOpeningStatus
+import com.infomaniak.mail.databinding.FragmentNewMessageBinding
+import com.infomaniak.mail.utils.UiUtils
+import com.infomaniak.mail.utils.observeNotNull
+import dagger.hilt.android.qualifiers.ActivityContext
+import dagger.hilt.android.scopes.ActivityScoped
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.math.roundToInt
+import com.infomaniak.lib.core.R as RCore
+
+@ActivityScoped
+class NewMessageAiManager @Inject constructor(
+    @ActivityContext private val activityContext: Context,
+    private val localSettings: LocalSettings,
+) : NewMessageManager() {
+
+    private val activity get() = activityContext as Activity
+
+    private var _aiViewModel: AiViewModel? = null
+    private val aiViewModel: AiViewModel get() = _aiViewModel!!
+
+    private val animationDuration by lazy { resources.getInteger(R.integer.aiPromptAnimationDuration).toLong() }
+    private val scrimOpacity by lazy { ResourcesCompat.getFloat(context.resources, R.dimen.scrimOpacity) }
+    private val backgroundColor by lazy { context.getColor(R.color.backgroundColor) }
+    private val black by lazy { context.getColor(RCore.color.black) }
+
+    private var aiPromptFragment: AiPromptFragment? = null
```

Comment:
Could be something like this to avoid the !! operator:

```
private val aiPromptFragment by lazy {
        childFragmentManager.findFragmentByTag(AI_PROMPT_FRAGMENT_TAG) as? AiPromptFragment ?: AiPromptFragment()
    }
```

Reply:
`aiPromptFragment` needs to be set to `null` inside `removeFragmentAndHideScrim()` anyway so I'm afraid I needs to stay nullable.

Also, the lazy value wouldn't work because when calling `childFragmentManager.findFragmentByTag()` the value can change between two calls and caching its with a `by lazy` value would prevent it from ever updating